### ISSUE TITLE
[New Rule] Don't check list length is > 0 before iterating over it

### ISSUE THREAD

**Issue by rpdelaney** [MartinThoma/flake8-simplify#160]
## Explanation

Looking at one's own old, bad code can be a source of great cringe.

- I remember prepending `if incidents is not None` because if incidents _is_ None, then a `TypeError` is raised on the second condition. I suspect this line can be reduced to `if incidents:` without any change in behavior.

- Checking the list length before iterating over it is a smell. 

## Example

```python
# Bad
incidents = get_ticket_ids(cell.value)
if incidents is not None and len(incidents) > 0:
    for incident in incidents:
        print(incident)

# Good
incidents = get_ticket_ids(cell.value)
if incidents:
    for incident in incidents:
        print(incident)

# Good
if incidents := get_ticket_ids(cell.value):
    for incident in incidents:
        print(incident)
```

**Comment by rpdelaney** [comment]
Prior art:
* [PIE787](https://github.com/sbdchd/flake8-pie#pie787-no-len-condition)
